oci/client/pull.go

```diff
@@ -61,14 +103,72 @@ func (c *Client) Pull(ctx context.Context, url, outDir string) (*Metadata, error
 		return nil, fmt.Errorf("no layers found in artifact")
 	}
 
-	blob, err := layers[0].Compressed()
+	if len(layers) < o.layerIndex+1 {
+		return nil, fmt.Errorf("index '%d' out of bound for '%d' layers in artifact", o.layerIndex, len(layers))
+	}
+
+	err = extractLayer(layers[o.layerIndex], outPath, o)
 	if err != nil {
-		return nil, fmt.Errorf("extracting first layer failed: %w", err)
+		return nil, err
 	}
+	return meta, nil
+}
 
-	if err = tar.Untar(blob, outDir, tar.WithMaxUntarSize(-1), tar.WithSkipSymlinks()); err != nil {
-		return nil, fmt.Errorf("failed to untar first layer: %w", err)
+// extractLayer extracts the Layer to the path
+func extractLayer(layer gcrv1.Layer, path string, opts *PullOptions) error {
```

Comment:
Because we update the layer type in the function if it isn't specified (although this isn't strictly needed, we could use another variable) but I think we should leave the function as-is so we can persist the layertype in the opts struct

Reply:
thinking more about this, maybe we can do away with the extractLayer function and have this check in `Pull` instead.